src/screens/TaskDetailScreen.tsx

```diff
@@ -63,12 +99,24 @@ const TaskDetailScreen: React.FC<TaskDetailScreenProps> = ({ navigation, route }
   // Local state to manage the task
   const [task, setTask] = useState<Task>(initialTask);
 
+  // Animation values
+  const fadeAnim = useState(new Animated.Value(0))[0];
+
+  // Start fade-in animation when component mounts
+  React.useEffect(() => {
+    Animated.timing(fadeAnim, {
+      toValue: 1,
+      duration: 300,
+      useNativeDriver: true
+    }).start();
+  }, []);
+
   /**
    * Toggles task status between 'pending' and 'completed'
    * Updates both local state and parent component
    * Provides an undo option via alert
    */
-  const handleToggleStatus = () => {
+  const handleToggleStatus = useCallback(() => {
```

Comment:
2. Handler Closures & Undo Logic
Capturing the right “previous” state
In handleToggleStatus, your “Undo” callback uses the task value from the closure. That works now, but could become confusing if you refactor. A more explicit pattern is to store the old status in a local variable:

ts
Copy
Edit
const oldStatus = task.status;
// …
Alert.alert(…,
  [
    { … },
    {
      text: 'Undo',
      onPress: () => {
        const reverted = { …task, status: oldStatus };
        setTask(reverted);
        onUpdate(reverted);
      },
    }
  ]
);
This makes it crystal‑clear what you’re reverting to.